### PR TITLE
Update to net8

### PR DESCRIPTION
**Pull request by stefannikolei**
### Prerequisites

- [ ] I have written a descriptive pull-request title
- [ ] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [ ] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [ ] I have provided test coverage for my change (where applicable)

### CODE REVIEW

File: src/ImageSharp.ruleset [hunk 1]
```diff
@@ -1,7 +1,11 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <RuleSet Name="ImageSharp" ToolsVersion="17.0">
   <Include Path="..\shared-infrastructure\sixlabors.ruleset" Action="Default" />
+   <Rules AnalyzerId="Microsoft.CodeAnalysis.CSharp.NetAnalyzers" RuleNamespace="Microsoft.CodeAnalysis.CSharp.NetAnalyzers">
+    <Rule Id="CA1857" Action="None" />
```
[the remark this answers is not in the document]
Comment (JimBobSquarePants) [reply src/ImageSharp.ruleset:5]: Can we not make each method require a const then by decorating with the attribute?

**Comment by JimBobSquarePants** [comment]
Thanks for this. You might want to put a pause on it though until we ship v3.1 We'll need to update SharedInfrastructure submodule following that.

File: src/ImageSharp.ruleset (hunk 1 above)
[the remark this answers is not in the document]
Comment (stefannikolei) [reply src/ImageSharp.ruleset]: I missed that attribute. I added it.

**Comment by stefannikolei** [comment]
@JimBobSquarePants should we run the arm tests before we merge this?

**Comment by JimBobSquarePants** [comment]
@stefannikolei triggered.

File: src/ImageSharp/ImageSharp.csproj [hunk 2]
```diff
@@ -17,6 +17,7 @@
 
   <!--  This enables the nullable analysis and treats all nullable warnings as error-->
   <PropertyGroup>
+    <LangVersion>12.0</LangVersion>
```
Comment (stefannikolei) [inline src/ImageSharp/ImageSharp.csproj:20]: This can probably be removed when SharedInfrastructure was updated

**Comment by stefannikolei** [comment]
Microsoft.DotNet.RemoteExecutor I can not find nor update this package...

File: src/ImageSharp/Common/Helpers/Numerics.cs [hunk 3]
```diff
@@ -908,25 +908,6 @@ public static int ReduceSum(Vector256<int> accumulator)
         return Sse2.ConvertToInt32(vsum);
     }
 
-    /// <summary>
-    /// Reduces elements of the vector into one sum.
-    /// </summary>
-    /// <param name="accumulator">The accumulator to reduce.</param>
-    /// <returns>The sum of all elements.</returns>
-    [MethodImpl(InliningOptions.ShortMethod)]
-    public static int ReduceSumArm(Vector128<uint> accumulator)
```
[the remark this answers is not in the document]
Comment (stefannikolei) [reply src/ImageSharp/Common/Helpers/Numerics.cs:917]: yes

File: src/ImageSharp.ruleset [hunk 4]
```diff
@@ -1,7 +1,11 @@
...
+    <Rule Id="CA1859" Action="None" />
```
[the remark this answers is not in the document]
Comment (JimBobSquarePants) [reply src/ImageSharp.ruleset:5]: We should open up a discussion or issue to track it so it's not forgotten.

**Comment by stefannikolei** [comment]
I just ran the JpegEncoder tests against .NET6 and .NET8. Those look promising.

```




.NET8

BenchmarkDotNet v0.13.10, macOS Sonoma 14.1.2 (23B92) [Darwin 23.1.0]
Intel Core i9-9880H CPU 2.30GHz, 1 CPU, 16 logical and 8 physical cores
.NET SDK 8.0.100
  [Host]     : .NET 8.0.0 (8.0.23.53103), X64 RyuJIT AVX2
  DefaultJob : .NET 8.0.0 (8.0.23.53103), X64 RyuJIT AVX2


| Method                              | Mean      | Error     | StdDev    |
|------------------------------------ |----------:|----------:|----------:|
| 'Baseline 4:4:4 Interleaved'        | 11.185 ms | 0.3432 ms | 0.9957 ms |
| 'Baseline 4:2:0 Interleaved'        |  8.775 ms | 0.1736 ms | 0.4355 ms |
| 'Baseline 4:0:0 (grayscale)'        |  1.254 ms | 0.0173 ms | 0.0135 ms |
| 'Progressive 4:2:0 Non-Interleaved' | 11.796 ms | 0.2352 ms | 0.3297 ms |


.NET6

BenchmarkDotNet=v0.13.0, OS=macOS 14.1.2 (23B92) [Darwin 23.1.0]
Intel Core i9-9880H CPU 2.30GHz, 1 CPU, 16 logical and 8 physical cores
.NET SDK=8.0.100
  [Host]     : .NET 6.0.21 (6.0.2123.36311), X64 RyuJIT
  DefaultJob : .NET 6.0.21 (6.0.2123.36311), X64 RyuJIT


|                              Method |      Mean |     Error |    StdDev |
|------------------------------------ |----------:|----------:|----------:|
|        'Baseline 4:4:4 Interleaved' | 18.003 ms | 0.3533 ms | 0.6636 ms |
|        'Baseline 4:2:0 Interleaved' | 11.986 ms | 0.2339 ms | 0.3279 ms |
|        'Baseline 4:0:0 (grayscale)' |  2.181 ms | 0.0398 ms | 0.0426 ms |
| 'Progressive 4:2:0 Non-Interleaved' | 16.453 ms | 0.3239 ms | 0.4211 ms |
```

**Pull request merged**